.github/workflows/gradle.yml

```diff
@@ -51,6 +52,69 @@ jobs:
       - name: Setup Gradle
         uses: gradle/actions/setup-gradle@v4
 
+      - name: Setup WireGuard (ubuntu)
```

Comment:
I think you should probably move the wireguard setup into a dedicated workflow file and include it https://docs.github.com/en/actions/sharing-automations/reusing-workflows#calling-a-reusable-workflow

Reply:
I ended up creating a GitHub Action since it was the only way I found to make it work, let me know what you think!